arduino/src/arduino.cpp

```diff
@@ -0,0 +1,214 @@
+    serial.writeString(message.c_str());
+    // check sum should be added.
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+
+Arduino::ERROR_ARDUINO Arduino::writeFromBufferToFile(const std::string &data, int place)
+{
+    std::ofstream dataLog;
+    std::string fileName = FILE_NAME;
+    fileName.append(std::to_string(place));
+    fileName.append(FILE_NAME_END);
+    dataLog.open(fileName, std::ios_base::app);
+    if (!dataLog.is_open())
+    {
+        return Arduino::ERROR_ARDUINO::ERROR_WITH_FILES;
+    }
+
+    dataLog << std::time(nullptr) << std::endl;
+    dataLog << data << std::endl;
+    dataLog.close();
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
+
+Arduino::ERROR_ARDUINO Arduino::getDataWithWhileLoop()
+{
+    Arduino::ERROR_ARDUINO return_code = Arduino::ERROR_ARDUINO::SUCCSESS;
+    return_code = openSerial();
+    if (return_code != Arduino::ERROR_ARDUINO::SUCCSESS)
+    {
+        goto l_cleanup;
+    }
+
+    while (true)
+    {
+        if (serial.available())
+        {
+            uint32_t num = 0;
+            this->serial.readBytes(&num, sizeof(num));
+            switch (num)
+            {
+            case Arduino::HEADERS::GET_FLOAT:
+                return_code = receiveFloatFromSensor();
+                if (return_code != SUCCSESS)
+                    goto l_cleanup;
+                continue;
+            case Arduino::HEADERS::ASK_FOR_OVERALL_STATUS:
+                return_code = checkStatusFromArduino();
+                if (return_code != SUCCSESS)
+                    goto l_cleanup;
+                continue;
+            default:
+                continue;
+            }
+        }
+    }
+
+l_cleanup:
+    return return_code;
+}
+
+Arduino::ERROR_ARDUINO Arduino::receiveFloatFromSensor()
+{
+    Arduino::ERROR_ARDUINO errorCode = Arduino::ERROR_ARDUINO::SUCCSESS;
+    int id = receiveInt();
+    float f = receiveFloat();
+    errorCode = this->writeFromBufferToFile(std::to_string(f), id);
+    return errorCode;
+}
+
+float Arduino::receiveFloat()
+{
+    float value = 0;
+    serial.readBytes(&value, sizeof(value));
+    return value;
+}
+
+void Arduino::addLogFilesForSensors()
+{
+    addFile("..//dataFromArduino//sensor0.log");
+    addFile("..//dataFromArduino//sensor1.log");
+    addFile("..//dataFromArduino//sensor2.log");
+    addFile("..//dataFromArduino//sensor3.log");
+    addFile("..//dataFromArduino//sensor4.log");
+    addFile("..//dataFromArduino//sensor5.log");
+}
+
+bool Arduino::receiveBoolean()
+{
+    uint8_t value = 0;
+    serial.readBytes(&value, sizeof(value));
+    return value;
+}
+
+Arduino::ERROR_ARDUINO Arduino::checkStatusFromArduino()
+{
+    if (receiveBoolean())
+    {
+        std::cout << "the Status is great!" << std::endl;
+    }
+    else
+    {
+        std::cout << "the Status is bad!" << std::endl;
+        return Arduino::ERROR_ARDUINO::STATUS_IS_BAD;
+    }
+
+    return Arduino::ERROR_ARDUINO::SUCCSESS;
+}
```

Comment:
why reading bool false mean status is bad?

Reply:
Does it make sense?